unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -117,39 +117,34 @@ TEST_CASE("Syncer dynamic FPS - throughput test", "[live]")
         {
             if (frames_arrival_info.empty())
                 return;
-            float fps = 0.0f;
-            while (!frames_arrival_info.empty() && fps < 10) // skip unstable frames at the beginning
-            {
-                fps = frames_arrival_info.front().second;
-                frames_arrival_info.erase(frames_arrival_info.begin());
-            }
-            if (frames_arrival_info.size() < 10 || fps < 1.0f)
-                return;
             std::vector<std::pair<float, float>>  vec[2]; // separate frames according to fps (before and after setting exposure)
+            std::vector<std::pair<float, float>> extra_frames;
             for (auto& frm : frames_arrival_info)
             { 
-                auto fps_ratio = frm.second / fps; // used ratio instead of precised fps because syncer doesn't change fps to exact value when running RGB
-                if (fps_ratio > 0.9)
+                // used ratio instead of precised fps because syncer doesn't change fps to exact value when running RGB
+                if (frm.second / 60.0 > 0.9)
                     vec[0].push_back(frm);
-                else
+                else if (frm.second / 30.0 > 0.9)
                     vec[1].push_back(frm);
+                else
+                    extra_frames.push_back(frm);
             }
 
             int i = 0;
             for (auto& v : vec)
             {
-                if (v.empty())
+                if (v.size() < 10)
                     continue;
                 auto dt = (v.back().first - v.front().first)/ 1000000;
                 auto actual_fps = v.front().second;
                 float calc_fps = (float)v.size() / dt;
-                float fps_ratio = calc_fps / actual_fps;
+                float fps_ratio = abs(1 - calc_fps / actual_fps);
                 CAPTURE(stream_type, actual_fps, calc_fps, v.size());
-                CHECK(fps_ratio > 0.9);
-                CHECK(fps_ratio < 1.3);
+                CHECK(fps_ratio < 0.1);
                 check_frame_drops(v);
                 i += 1;
             }
+            CHECK(extra_frames.size() / frames_arrival_info.size() < 0.1);
```

Comment:
What are the nature of those frames? Why 10%

Reply:
those are unstable frames that received after changing exposure - I will add a comment